### PR TITLE
Remove deprecations that expire in 3.3

### PR DESCRIPTION
**Pull request by QuLogic**
## PR Summary

See title. There are still a few left, but they need different changes.

Also, sort the removal list (this looks a little bigger than it really is.)

## PR Checklist

- [x] Has Pytest style unit tests
- [x] Code is [Flake 8](http://flake8.pycqa.org/en/latest/) compliant
- [N/A] New features are documented, with examples if plot related
- [x] Documentation is sphinx and numpydoc compliant
- [N/A] Added an entry to doc/users/next_whats_new/ if major new feature (follow instructions in README.rst there)
- [x] Documented in doc/api/api_changes.rst if API changed in a backward-incompatible way

### CODE REVIEW

**Comment by QuLogic** [comment]
There's still a few left:
* `save_count` in animations; https://github.com/matplotlib/matplotlib/issues/8278#issuecomment-619288371
* #12430, but I'm not sure what @timhoffm wanted to do there; change to a keyword argument?
* all of `mpl_toolkits/axes_grid`; probably can do that, but it's been there since 2.2, and I don't know why we're keeping it around still

**Comment by timhoffm** [comment]
I will take care of #12430 in a separate PR.

File: lib/matplotlib/legend.py [hunk 1]
```diff
@@ -444,28 +444,15 @@ def __init__(self, parent, handles, labels,
                 loc = 'upper right'
         if isinstance(loc, str):
             if loc not in self.codes:
```
Comment (timhoffm) [inline lib/matplotlib/legend.py:446]: Would this now better be a `try except KeyError: raise ValueError`?

Comment (QuLogic) [reply lib/matplotlib/legend.py:446]: I think that just clutters the traceback:
```ipynb
In [2]: plt.figure().legend(loc='ca')
```
with the `try`, adds one more traceback frame, which is not really any clearer to anyone:
```pytb
KeyError                                  Traceback (most recent call last)
~/code/matplotlib/lib/matplotlib/legend.py in __init__(self, parent, handles, labels, loc, numpoints, markerscale, markerfirst, scatterpoints, scatteryoffsets, prop, fontsize, borderpad, labelspacing, handlelength, handleheight, handletextpad, borderaxespad, columnspacing, ncol, mode, fancybox, shadow, title, title_fontsize, framealpha, edgecolor, facecolor, bbox_to_anchor, bbox_transform, frameon, handler_map)
    446             try:
--> 447                 loc = self.codes[loc]
    448             except KeyError:

KeyError: 'ca'

During handling of the above exception, another exception occurred:

ValueError                                Traceback (most recent call last)
<ipython-input-2-c48871c74701> in <module>()
----> 1 plt.figure().legend(loc='ca')

~/code/matplotlib/lib/matplotlib/figure.py in legend(self, *args, **kwargs)
   1827             # extra_args = extra_args[1:]
   1828             pass
-> 1829         l = mlegend.Legend(self, handles, labels, *extra_args, **kwargs)
   1830         self.legends.append(l)
   1831         l._remove_method = self.legends.remove

~/code/matplotlib/lib/matplotlib/legend.py in __init__(self, parent, handles, labels, loc, numpoints, markerscale, markerfirst, scatterpoints, scatteryoffsets, prop, fontsize, borderpad, labelspacing, handlelength, handleheight, handletextpad, borderaxespad, columnspacing, ncol, mode, fancybox, shadow, title, title_fontsize, framealpha, edgecolor, facecolor, bbox_to_anchor, bbox_transform, frameon, handler_map)
    449                 raise ValueError(
    450                     "Unrecognized location {!r}. Valid locations are\n\t{}\n"
--> 451                     .format(loc, '\n\t'.join(self.codes)))
    452         if not self.isaxes and loc == 0:
    453             raise ValueError(

ValueError: Unrecognized location 'ca'. Valid locations are
	best
	upper right
	upper left
	lower left
	lower right
	right
	center left
	center right
	lower center
	upper center
	center
```

**Pull request merged**